woocommerce/woocommerce-ios
# Edit Coupon flow: editing of the description and of the coupon expiry date

Part of #6488

### Description
As part of the Editing Coupon Flow, I implemented some new logic and flow in the Edit Coupon screen.

- Implemented a new reusable SwiftUI component called `FullScreenTextView`, which is a view to input a simple text, and that was used to add the optional description to the coupon.
- Now we are presenting the Restrictions view, but without any binding between the views for the moment.
- Now it’s possible to edit the coupon expiry date, using a new component called `CouponExpiryDateView` (that eventually we can rename to be reusable). I implemented it from scratch because we don’t have any view for selecting a single date, we just used a complex navigation flow in order filters for selecting a range, but I believe this simple view it’s better.
- I will allocate the `AddEditCouponViewModel` in the `CouponDetails`, since allocating it every time in the sheet will have an [unaxpected behavior](https://a8c.slack.com/archives/C02Q9GACF/p1649408989455409).

Added @toupper between reviewers as part of his onboarding process.

### Testing instructions
1. Open a coupon detail and make sure the feature flag for editing a coupon is enabled.
2. Press the ellipsis menu in the navigation menu.
3. Try to edit a description of a coupon, you should see a new view pushed where you can edit the text.
4. Try to edit or delete the expiry date, you should see a new view pushed with the date picker.
5. Press the "Usage Restrictions" row, and you should see the view pushed.


### Screenshots
| Edit Description screen             |  Select expiry date |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 13 Pro Max - 2022-04-08 at 16 52 43](https://user-images.githubusercontent.com/495617/162464577-5ea6cf74-8494-4943-9580-0a0affa51af6.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2022-04-08 at 16 53 17](https://user-images.githubusercontent.com/495617/162464715-e98eea1e-6463-4351-a887-6d8af45bf4e2.png)



---
